### PR TITLE
Modernize and prepare for 1.4.0 release

### LINKED ISSUE #57: next release ETA
Hello, I see that you have prepared commit merged in master about python3.10 support.
When do you plan to release the version including this [commit](https://github.com/pytest-dev/pytest-forked/commit/90b141fbb9206c17b36ff16b08d111adeed12f74)?
Thanks

### PR DESCRIPTION
**Pull request by nicoddemus**
* Fix CI and add badges
* Remove Travis
* Update CHANGELOG for 1.4.0
* Add pre-commit and black
* Add RELEASING docs

After this is done we can make the release directly from this branch.

Close #57

### CODE REVIEW

File: testing/test_xfail_behavior.py
Comment (RonnyPfannschmidt) [inline testing/test_xfail_behavior.py:118]: f string?

Comment (nicoddemus) [reply testing/test_xfail_behavior.py]: Fixed. I also added `pyupgrade` to the pre-commit config.

File: .pre-commit-config.yaml
Comment (webknjaz) [inline .pre-commit-config.yaml:10]: Is it possible to avoid adding this?

Comment (nicoddemus) [reply .pre-commit-config.yaml:10]: It is, but why?

Comment (webknjaz) [reply .pre-commit-config.yaml:10]: Bad prior experience, gives me flashbacks to the point I try to avoid this thing/contribute to projects that try to throw it in my face.

Comment (nicoddemus) [reply .pre-commit-config.yaml:10]: Hmm sorry to hear that. In my experience this is always beneficial, because both contributors and reviewers no longer need to worry about code formatting at all.

I personally prefer to have it, but I will let @RonnyPfannschmidt make the final call.

Comment (RonnyPfannschmidt) [reply .pre-commit-config.yaml:10]: anything on pytest-dev we use brings in black/pre-commit - so its pretty much our standard already - its not like there will be  any surprises afterwards unless one tries to break them and/or deliberately avoids running them

**Comment by nicoddemus** [comment]
(Just to note that this is waiting for approval to move forward).

**Review by RonnyPfannschmidt — APPROVED** [review]

**Pull request merged**
